### PR TITLE
Faster pareto front calculation

### PR DESCRIPTION
**Pull request by uri-granta**

### CODE REVIEW

File: trieste/acquisition/multi_objective/dominance.py [hunk 1]
```diff
@@ -15,30 +15,46 @@
...
-    )
-
-    return tf.boolean_mask(observations, dominance == 0), dominance
+    orig_points = observations.shape[0]
+    orig_indices = np.arange(orig_points)
+
+    if orig_points > 0:
+        # Reordering the observations beforehand speeds up the search
+        obs_np = to_numpy(observations)
+        idx = np.argsort(((obs_np - obs_np.mean(axis=0)) / (obs_np.std(axis=0) + 1e-7)).sum(axis=1))
```
Comment (hstojic) [inline trieste/acquisition/multi_objective/dominance.py:44]: will this trigger retracing if this function is used anywhere in tf.function?

**Review by hstojic — APPROVED** [review]
looks good, the only concern is that perhaps numpy might trigger retracing?
but perhaps non_dominated is not used anywhere inside tf.function?

File: trieste/acquisition/multi_objective/dominance.py (hunk 1 above)
Comment (uri-granta) [reply trieste/acquisition/multi_objective/dominance.py:44]: Very good spot! In fact the use of not just `to_numpy` but also `np.any` and `np.sum` in the main loop all fail inside a tf.function! I'll have a go at rewriting this in pure tensorflow.

Comment (uri-granta) [reply trieste/acquisition/multi_objective/dominance.py:44]: I've added a test, but fixing this will probably be a pain :(

**Pull request closed without merging**